src/openms/include/OpenMS/CONCEPT/ProgressLogger.h

```diff
@@ -100,12 +104,13 @@ namespace OpenMS
     mutable time_t last_invoke_;
     static int recursion_depth_;
 
-    /// Return the name of the factory product used for this log type
-    static String logTypeToFactoryName_(LogType type);
-
     mutable ProgressLoggerImpl* current_logger_;
 
   };
 
+  // Function pointer for injecting the GUI progress logger implementation
+  typedef ProgressLogger::ProgressLoggerImpl* (*MakeGUIProgressLoggerFunc)();
+  extern MakeGUIProgressLoggerFunc make_gui_progress_logger;
```

Comment:
here I define the factory. by default it returns the NoProgressLogger. can be overwritten by OPENMS_GUI.lib to provide a Qt based logger GUIProgressLogger